src/arrayMethodReduce.js

```diff
@@ -4,8 +4,21 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(...args) {
+    const [callback, initialValue] = args;
+    let startIndex = 0;
+    let result = initialValue;
+
+    if (args.length === 1) {
+      startIndex = 1;
+      result = this[0];
+    }
```

Comment:
You could use `arguments` to access the array of arguments instead of using destructuring. Additionally, you can use a ternary operator to set the values of `startIndex` and `result` variables✨